models/repo/setting.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"code.gitea.io/gitea/models/db"
+
+	"xorm.io/builder"
+)
+
+// Setting is a key value store of repo settings
+type Setting struct {
+	ID           int64  `xorm:"pk autoincr"`
+	RepoID       int64  `xorm:"index unique(key_repoid)"`              // to load all of someone's settings
+	SettingKey   string `xorm:"varchar(255) index unique(key_repoid)"` // ensure key is always lowercase
+	SettingValue string `xorm:"text"`
+}
+
+// TableName sets the table name for the settings struct
+func (s *Setting) TableName() string {
+	return "repo_setting"
+}
+
+func init() {
+	db.RegisterModel(new(Setting))
+}
+
+// GetRepoSettings returns specific settings from repo
+func GetRepoSettings(rid int64, keys []string) (map[string]*Setting, error) {
+	settings := make([]*Setting, 0, len(keys))
+	if err := db.GetEngine(db.DefaultContext).
+		Where("repo_id=?", rid).
+		And(builder.In("setting_key", keys)).
+		Find(&settings); err != nil {
+		return nil, err
+	}
+	settingsMap := make(map[string]*Setting)
+	for _, s := range settings {
+		settingsMap[s.SettingKey] = s
+	}
+	return settingsMap, nil
+}
+
+// GetRepoAllSettings returns all settings from repo
+func GetRepoAllSettings(rid int64) (map[string]*Setting, error) {
+	settings := make([]*Setting, 0, 5)
+	if err := db.GetEngine(db.DefaultContext).
+		Where("repo_id=?", rid).
+		Find(&settings); err != nil {
```

Comment:
```suggestion
	if err := db.GetEngine(db.DefaultContext).Find(&settings, &Setting{RepoID: rid}); err != nil {
```
(Or adapted to the changes proposed below:)
```suggestion
	if err := db.GetEngine(db.DefaultContext).Find(&settings, &Setting{RepoID: repoID}); err != nil {
```